thunder/tests/test_grad.py

```diff
@@ -1218,6 +1237,7 @@ def upcast_tensors(x: Any) -> Any:
         f for f in reference_flats if isinstance(f, torch.Tensor) and f.requires_grad
     )
     reference_result = torch_op(*reference_args, **reference_kwargs)
+    reference_result = filter_differentiable_outputs(reference_result)
     reference_grad_result = torch.autograd.grad(reference_result, reference_tensors_requiring_grad, grads)
```

Comment:
Shouldn't we check that `len(grads) == len(reference_result)`? Or are we fine with `torch.autograd.grad` complaining?